logstash-core/lib/logstash/pipeline.rb

```diff
@@ -194,6 +197,7 @@ def run
     shutdown_workers
 
     @filter_queue_client.close
+    @queue.close
```

Comment:
re-reading the code I do not think the client should actually close the queue, here's why:
- the queue object returned by `build_queue_from_settings` is the "main" object from which the read and write clients are created. 
- there are 2 clients from the same queue object, which one do you close and when? to me the "client" should "close" itself but not necessarily close its parent object so that, for example, the write side could close itself but the read side could still be opened. 
- it would be easier to understand to uniformly support close on both the read and write clients which does not necessarily affect the queue object itself and leave the queue closing explicit. It make the closing logic easier to understand and follow IMO.

So I'd actually vote to add close on both clients and keep queue closing as-is.